prog/dftb+/lib_timedep/linrespgrad.F90

```diff
@@ -2034,6 +2035,10 @@ subroutine writeExcitations(sym, osz, nexc, nmatup, getij, win, eval, evec, wij,
       if (.not.tDegenerate) then
         call taggedWriter%write(fdTagged, tagLabels%excEgy, eval)
         call taggedWriter%write(fdTagged, tagLabels%excOsc, osz)
+        if (fdTradip > 0) then
```

Comment:
That's a risky check. If units are auto-generated (with ``newunit=``), they will be negative according to the standard.

Reply:
Yes, the whole file and linresp.F90 both need to be refactored to remove generated file id as these are used for _all_ output files.

Reply:
OK, I see, it's all over in the file :disappointed: